src/neo/Ledger/Blockchain.cs

```diff
@@ -459,7 +460,7 @@ private void Persist(Block block)
                     clonedSnapshot.Transactions.Add(tx.Hash, state);
                     clonedSnapshot.Transactions.Commit();
 
-                    using (ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Application, tx, clonedSnapshot, tx.SystemFee))
+                    using (ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Application, tx, clonedSnapshot, tx.SystemFee * ratio))
```

Comment:
Very good. I like this design.
Thus, ratio 0.5 makes the network more expensive.